aws_lambda_builders/workflow.py

```diff
@@ -40,25 +40,34 @@ def sanitize(func):
 
     @functools.wraps(func)
     def wrapper(self, *args, **kwargs):
-        valid_paths = []
+        valid_paths = {}
+        invalid_paths = {}
         # NOTE: we need to access binaries to get paths and resolvers, before validating.
         binaries_copy = self.binaries
         for binary, binary_path in binaries_copy.items():
+            invalid_paths[binary] = []
             validator = binary_path.validator
             exec_paths = binary_path.resolver.exec_paths if not binary_path.path_provided else binary_path.binary_path
             for executable_path in exec_paths:
-                valid_path = None
                 try:
                     valid_path = validator.validate(executable_path)
+                    if valid_path:
+                        valid_paths[binary] = valid_path
```

Comment:
Is `valid_paths` value of a key only a `str` and not a list, like `invalid_paths` because we consider only one path to be valid and end the search?

Reply:
Yes, if we encounter a valid path. we end the search.